File: front/src/models/Materiel.ts
export type Materiel = 
{
    kind: "Materiel",
    id: number,
    type: TypeMateriel,
    nom: string,
    description: string | null,
    prix: number,
    stock: number,
    nbPlacer: number,
    nbDetruit: number
}

export type MaterielRequete = 
{
    idTypeMateriel: number,
    nom: string,
    description: string | null,
    prix: number,
    stock: number,
    nbPlacer: number,
    nbDetruit: number
}

export type TypeMateriel = 
{
    id: number,
    nom: string
}

File: back/back/Models/Materiel.cs
using LiteDB;

namespace back.Models;

public sealed class Materiel
{
     [BsonId]
     public int Id { get; set; }

     [BsonRef]
     public MaterielType Type { get; set; } = null!;
     public string Nom { get; set; } = null!;
     public string? Description { get; set; }
     public int Prix { get; set; }
     public int Stock { get; set; }
     public int NbPlacer { get; set; }
     public int NbDetruit { get; set; }
}
